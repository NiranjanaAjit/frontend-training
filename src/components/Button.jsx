import React from "react";
const Button = ({ text, buttonClassName, handleSubmit }) => {
  return (
    <button
      type="submit"
      data-testid="button-test-id"
      className={buttonClassName}
      onClick={(e) => handleSubmit(e)}
    >
      {text}
    </button>
  );
};

export default Button;
