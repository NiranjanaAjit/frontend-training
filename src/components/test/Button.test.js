import React from "react";
import { fireEvent, render } from "@testing-library/react";
import Button from "../Button";
describe("Button Test Cases", () => {
  test("Rendering button", () => {
    const {getByTestId} = render(<Button/>)
    const element = getByTestId('button-test-id');
    // console.log(element);
    expect(element).toBeTruthy()
  });

  test("Check if text is correct", () => {
    const text="click"
    const {getByText} = render(<Button text={text}/>)
    getByText(text);
  });

  test("Check if onClick is triggered", () => {
    const onClick = jest.fn();
    const {getByTestId} = render(<Button handleSubmit={onClick}/>)
    const element = getByTestId('button-test-id');  
    fireEvent.click(element);
    expect(onClick).toHaveBeenCalled();
});

test("Check if snapshots are mapped properly", () => {
    const onClick = jest.fn();
    const text = "bye"
    const {asFragment} = render(<Button text={text}  handleSubmit={onClick}/>)
    expect(asFragment()).toMatchSnapshot();
  });

});
