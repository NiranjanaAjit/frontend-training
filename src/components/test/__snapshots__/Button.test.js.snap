// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test Cases Check if snapshots are mapped properly 1`] = `
<DocumentFragment>
  <button
    data-testid="button-test-id"
    type="submit"
  >
    bye
  </button>
</DocumentFragment>
`;
